guides/runtime-parser.md

```diff
@@ -0,0 +1,308 @@
+# Introduction
+
+*This guide assumes that the [App Shell guide](https://mobile.angular.io/guides/app-shell.html) has been completed.*
+
+Now we're going to take a look at another part of the Angular Mobile Toolkit - the App Shell Runtime Parser. The Runtime Parser is a library which has the following features:
+
+- Automatically generates the App Shell of our application by using a predefined template.
+- Provides the App Shell template from the local cache automatically for each view that should use it.
+
+This way, when the user navigates to a different view in the application she will instantly get a minimal working UI that will be later replaced with the view's full functionality. You can find more about the App Shell concept in the context of [Progressive Web Applications](https://developers.google.com/web/progressive-web-apps/) (PWA) on the [following link](https://developers.google.com/web/updates/2015/11/app-shell?hl=en).
+
+# Introduction to the Runtime Parser
+
+The Angular App Shell Runtime Parser works together with the App Shell directives, described in the [previous section](https://mobile.angular.io/guides/app-shell.html), and [Angular Universal](https://universal.angular.io).
+
+Now lets dig deeper in the features of the parser!
+
+First, lets suppose we have the following "Hello Mobile" component:
+
+```typescript
+import { Component } from '@angular/core';
+import { APP_SHELL_DIRECTIVES } from '@angular/app-shell';
+import { MdToolbar } from '@angular2-material/toolbar';
+import { MdSpinner } from '@angular2-material/progress-circle';
+
+@Component({
+  moduleId: module.id,
+  selector: 'hello-mobile-app',
+  template: `
+    <md-toolbar>
+      <div class="icon ng"></div>
+      {{title}}
+    </md-toolbar>
+    <md-spinner *shellRender></md-spinner>
+    <h1 *shellNoRender>App is Fully Rendered</h1>
+  `,
+  styles: [`
+    md-spinner {
+      margin: 24px auto 0;
+    }
+    .icon {
+      width: 40px;
+      height: 40px;
+      display: inline-block;
+    }
+    .icon.ng {
+      background-image: url(./images/angular.png);
+    }
+  `],
+  directives: [APP_SHELL_DIRECTIVES, MdToolbar, MdSpinner]
+})
+export class HelloMobileAppComponent {
+  title = 'Hello Mobile';
+}
+```
+
+...and in Universal, we bootstrap our application in the following way:
```

Comment:
Since this is the first time any of our guides are demonstrating a configuration for runtime prerendering, it should be called out that this is new stuff, similar to what we have for the app shell configuration, but a little bit different. Giving this module a descriptive filename is one easy way to set it apart from the app shell config.